lib/maths/CDecompositionComponent.cc

```diff
@@ -136,6 +137,21 @@ TDoubleDoublePr CDecompositionComponent::value(double offset, double n, double c
             return {m, m};
         }
 
```

Comment:
One way for `m` to be `nan` is if `offset` is `nan` when the function is called.  It could be worth checking if `offset` is `nan` before the line that checks `if (this->initialized()) {`.